pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go

```diff
@@ -12,6 +12,7 @@ import (
 // OpenShiftClustersClientAddons contains addons for OpenShiftClustersClient
 type OpenShiftClustersClientAddons interface {
 	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230904.OpenShiftCluster) error
+	UpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230904.OpenShiftClusterUpdate) error
```

Comment:
We don't need to add this, these clients are only used internally, nothing would call this.

Reply:
Yes that makes sense I'll remove them from `20230904` but will keep on other two where it is needed `20231122` and `20240812preview`.

Reply:
Fixed U+1F44D

Reply:
Plus the question with this pr or to include it with e2e test suite work, I have another branch for my e2e test suite work which I could bump it off after this PR later with the help of rebase. If it's fine with everyone.

Reply:
No worries made this whole pr focusing towards e2e test suite work plus including the missing functionality to our internal utility clients. Basically all in one pr. Therefore closing this conversation with resolved mark.